latch/resources/workflow.py

```diff
@@ -1,95 +1,104 @@
 import inspect
 from dataclasses import is_dataclass
 from textwrap import dedent
-from typing import Callable, Union, get_args, get_origin
+from typing import Callable, Dict, Optional, Union, get_args, get_origin
 
+import click
 from flytekit import workflow as _workflow
 from flytekit.core.workflow import PythonFunctionWorkflow
 
-from latch.types.metadata import LatchMetadata
+from latch.types.metadata import LatchAuthor, LatchMetadata, LatchParameter
+from latch_cli.utils import best_effort_display_name
+
+
+def _generate_metadata(f: Callable) -> Dict[str, LatchParameter]:
+    signature = inspect.signature(f)
+    metadata = LatchMetadata(f.__name__, LatchAuthor())
+    metadata.parameters = {
+        param: LatchParameter(display_name=best_effort_display_name(param))
+        for param in signature.parameters
+    }
+    return metadata
+
+
+def _inject_metadata(f: Callable, metadata: LatchMetadata) -> None:
+    if f.__doc__ is None:
+        f.__doc__ = f"{f.__name__}\n\nSample Description"
+    short_desc, long_desc = f.__doc__.split("\n", 1)
+    f.__doc__ = f"{short_desc}\n{dedent(long_desc)}\n\n" + str(metadata)
 
 
 # this weird Union thing is to ensure backwards compatibility,
 # so that when users call @workflow without any arguments or
 # parentheses, the workflow still serializes as expected
 def workflow(
-    metadata: Union[LatchMetadata, Callable]
+    metadata: Optional[Union[LatchMetadata, Callable]] = None
```

Comment:
```suggestion
    metadata: Union[LatchMetadata, Callable]
```

i like disallowing `@workflow()` altogether as it makes everything more unambiguous